shared/src/main/java/io/crate/concurrent/CompletableFutures.java

```diff
@@ -57,4 +67,349 @@ private CompletableFutures() {
         }
     }
 
+    public static <T> CompletionStage<T> asKillable(CompletableFuture<T> delegate,
+                                                    Consumer<Throwable> beforeExceptional) {
+        return new KillableCompletableFuture<>(delegate, beforeExceptional);
+    }
+
+    @SuppressForbidden
```

Comment:
Just curious, what's the forbidden part here?

Reply:
See https://github.com/crate/crate/blob/master/gradle/forbidden-signatures.txt#L5